primary/src/primary.rs

```diff
@@ -103,6 +104,59 @@ impl Primary {
             channel(CHANNEL_CAPACITY);
         let (tx_state_handler, rx_state_handler) = channel(CHANNEL_CAPACITY);
 
+        // Monitor of channel capacity
+        let mon_tx_others_digests = tx_others_digests.clone();
+        let mon_tx_our_digests = tx_our_digests.clone();
+        let mon_tx_parents = tx_parents.clone();
+        let mon_tx_headers = tx_headers.clone();
+        let mon_tx_sync_headers = tx_sync_headers.clone();
+        let mon_tx_sync_certificates = tx_sync_certificates.clone();
+        let mon_tx_headers_loopback = tx_headers_loopback.clone();
+        let mon_tx_certificates_loopback = tx_certificates_loopback.clone();
+        let mon_tx_primary_messages = tx_primary_messages.clone();
+        let mon_tx_helper_requests = tx_helper_requests.clone();
+        let mon_tx_get_block_commands = tx_get_block_commands.clone();
+        let mon_tx_batches = tx_batches.clone();
+        let mon_tx_block_removal_commands = tx_block_removal_commands.clone();
+        let mon_tx_batch_removal = tx_batch_removal.clone();
+        let mon_tx_block_synchronizer_commands = tx_block_synchronizer_commands.clone();
+        let mon_tx_certificate_responses = tx_certificate_responses.clone();
+        let mon_tx_payload_availability_responses = tx_payload_availability_responses.clone();
+        let mon_tx_state_handler = tx_state_handler.clone();
+
+        tokio::task::spawn(async move {
```

Comment:
This doesn't work well: you want to collect the task created in this line, and put that handle (like all the task handles created in this file) in the `handles` vector at the end of this function.

Reply:
And also need too shutdown if told so. New patch incoming.